src/basic/fs-util.h

```diff
@@ -44,6 +44,8 @@ static inline int fchmod_and_chown(int fd, mode_t mode, uid_t uid, gid_t gid) {
 int fchmod_umask(int fd, mode_t mode);
 int fchmod_opath(int fd, mode_t m);
 
+int fchmodat_best(int dirfd, const char *path, mode_t mode, int flags);
```

Comment:
Bikeshedding: this is not very descriptive. Maybe call this "fchmodat2_graceful" or something?